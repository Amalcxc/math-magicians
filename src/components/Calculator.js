import React, { Component } from 'react';
import Calculate from '../logic/calculate.js';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
    };
  }

  calculate = (e) => {
    this.setState((prev) => Calculate(prev, e.target.name));
  };

  render() {
    let input;
    const { total, next } = this.state;
    if (next === null && total === null) {
      input = <p>0</p>;
    } else if (next === null) {
      input = <p>{total}</p>;
    } else {
      input = <p>{next}</p>;
    }
    return <div className='calculator'>
      <div className='input'>
        {input}
      </div>
      <div className='calculator-buttons'>
        <button name='AC' type='button' className='cal-style' onClick={this.calculate}>AC</button>
        <button name='+/-' type='button' className='cal-style' onClick={this.calculate}>+/-</button>
        <button name='%' type='button' className='cal-style' onClick={this.calculate}>%</button>
        <button name='รท' type='button' className='cal-style is-orange' onClick={this.calculate}>รท</button>
        <button name='7' type='button' className='cal-style' onClick={this.calculate}>7</button>
        <button name='8' type='button' className='cal-style' onClick={this.calculate}>8</button>
        <button name='9' type='button' className='cal-style' onClick={this.calculate}>9</button>
        <button name='x' type='button' className='cal-style is-orange' onClick={this.calculate}>x</button>
        <button name='4' type='button' className='cal-style' onClick={this.calculate}>4</button>
        <button name='5' type='button' className='cal-style' onClick={this.calculate}>5</button>
        <button name='6' type='button' className='cal-style' onClick={this.calculate}>6</button>
        <button name='-' type='button' className='cal-style is-orange' onClick={this.calculate}>-</button>
        <button name='1' type='button' className='cal-style' onClick={this.calculate}>1</button>
        <button name='2' type='button' className='cal-style' onClick={this.calculate}>2</button>
        <button name='3' type='button' className='cal-style' onClick={this.calculate}>3</button>
        <button name='+' type='button' className='cal-style is-orange' onClick={this.calculate}>+</button>
        <button name='0' type='button' className='cal-style is-zero' onClick={this.calculate}>0</button>
        <button name='.' type='button' className='cal-style' onClick={this.calculate}>.</button>
        <button name='=' type='button' className='cal-style is-orange' onClick={this.calculate}>=</button>
      </div>
    </div>;
  }
}

export default Calculator;