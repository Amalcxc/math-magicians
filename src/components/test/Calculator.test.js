import Calculator from '../Calculator';
import React from 'react';
import { render, screen } from '@testing-library/react';

test('Renders Calculator Component', () => {
  render(<Calculator />);
  const calculatorAc = screen.getByText('Lets do Some Math!');
  expect(calculatorAc).toMatchSnapshot();
});
