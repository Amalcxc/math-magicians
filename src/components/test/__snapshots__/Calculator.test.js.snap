// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Calculator Component 1`] = `
<h4
  class="cal-text"
>
  Lets do Some Math!
</h4>
`;
